### ISSUE TITLE
CNN for CIFAR_10

### ISSUE THREAD

**Issue by OppiHmiK** [OppiHmiK/Artificial-Intelligence#3]
from keras.datasets import cifar10
from keras.utils import np_utils
from keras.models import Sequential as seq
from keras.layers.core import Dense,Dropout,Activation, Flatten
from keras.layers.convolutional import Conv2D, MaxPooling2D
from keras.optimizers import SGD, Adam, RMSprop
import matplotlib.pyplot as plt

img_cha = 3
img_r = 32; img_c = 32

bat_size = 128
epoch = 20
classes = 10
verbose = 1
val_split = 0.2
opti = RMSprop()

# NOTE : CIFAR 리퍼지토리에서 임의로 10개의 클래스를 뽑는 데이터 셋
(i_train, t_train), (i_test, t_test) = cifar10.load_data()

# NOTE : 타겟 데이터 원 핫 인코딩
tar_train = np_utils.to_categorical(t_train, classes)
tar_test = np_utils.to_categorical(t_test, classes)

# NOTE : 입력 데이터 정규화
i_train = i_train.astype('float32')
i_train /= 255
i_test = i_test.astype('float32')
i_test /= 255

# NOTE : 신경망 구축
model = seq()
model.add(Conv2D(32, (3,3), padding = 'same', input_shape = (img_r, img_c, img_cha)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2,2)))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(classes))
model.add(Activation('softmax'))

# NOTE : 학습
model.compile(loss = 'categorical_crossentropy', optimizer = opti, metrics = ['accuracy'])

hist = model.fit(i_train, tar_train, batch_size=bat_size,
                 epochs=epoch, validation_split=val_split, verbose=verbose)

print("학습중... ")

score = model.evaluate(i_test, tar_test, batch_size=bat_size,
                       verbose=verbose)

print("\n 결과 점수 : ", score[0])
print("결과 정확도 : ", score[1])

# NOTE : 모델 저장
model_json = model.to_json()
open('cifar10.json', 'w').write(model_json)
model.save_weights('cifar_weights.h5', overwrite= True)
